internal/proxy/search_util.go

```diff
@@ -200,6 +205,7 @@ func parseSearchInfo(searchParamsPair []*commonpb.KeyValuePair, schema *schemapb
 			GroupByFieldId:  groupByFieldId,
 			GroupSize:       groupSize,
 			StrictGroupSize: strictGroupSize,
+			Hints:           hints,
```

Comment:
better to not fill this member if hints equal empty. 
You can change this in next pr .